### PR TITLE
Add doctests to Channel methods

### PR DESCRIPTION
**Pull request by glennmoy**
Happy to keep going in case there are any more we'd like to add to.

### CODE REVIEW

File: base/channels.jl [hunk 1]
```diff
@@ -380,8 +380,37 @@ end
...
+# Examples
+
+Buffered channel:
+```jldoctest
+julia> c = Channel(3) do ch
+    foreach(i -> put!(ch, i), 1:4)
+end;
+
+julia> fetch(c)
+1
```
Comment (glennmoy) [inline base/channels.jl:395]: I could probably show that after fetching the item is not removed?

Comment (kshyatt) [reply base/channels.jl:395]: Sure, if you want to add a line with `collect`

File: base/channels.jl [hunk 2]
```diff
@@ -402,10 +431,33 @@ fetch_unbuffered(c::Channel) = throw(ErrorException("`fetch` is not supported on
...
+```jldoctest
+julia> c = Channel(0);
+
+julia> task = Task(() -> put!(c, 1));
+
+julia> schedule(task);
+
+julia> take!(c)
+1
+```
```
Comment (kshyatt) [inline base/channels.jl:459]: Double ```?

File: base/channels.jl (hunk 1 above)
Comment (glennmoy) [reply base/channels.jl:395]: added

File: base/channels.jl (hunk 2 above)
Comment (glennmoy) [reply base/channels.jl:459]: removed

**Comment by glennmoy** [comment]
running into this with the `wait` doctest https://github.com/JuliaDocs/Documenter.jl/issues/1580

**Comment by ViralBShah** [comment]
Is this ready to merge?

**Comment by glennmoy** [comment]
> Is this ready to merge?

yep! just waiting approval

**Comment by ViralBShah** [comment]
I'll wait for @kshyatt or someone to merge, but if nobody does, I'm happy to merge this on Monday (if you don't mind pinging me again).

**Pull request merged**